ui/client/reducers/graph.js

```diff
@@ -162,20 +164,25 @@ export function reducer(state, action) {
     }
     case "NODES_CONNECTED": {
       const edge = createEdge(action.fromNode, action.toNode, action.edgeType, state.processToDisplay.edges, action.processDefinitionData)
+      const newEdges = _.concat(state.processToDisplay.edges, edge)
       return {
         ...state,
         processToDisplay: {
           ...state.processToDisplay,
-          edges: _.concat(state.processToDisplay.edges, edge),
+          nodes: state.processToDisplay.nodes.map(n => action.toNode.id === n.id ?
+            enrichNodeWithProcessDependentData(action.toNode, action.processDefinitionData, newEdges) : n),
```

Comment:
From my observation `action.toNode` is not the correct one to take (it hasn't got `branchParameters`). Everything works fine with `n`